### PR TITLE
[5.2] Str: speed-up parseCallback

### PR DESCRIPTION
**Pull request by lucasmichot**

### CODE REVIEW

**Comment by taylorotwell** [comment]
What were the performance gains?

**Comment by donnysim** [comment]
The performance gain comes from not having to call an additional function `contains`, from a small test of 2000 iterations (local dev machine):

Old: 200ms 
New: 110ms

but if you inline `contains` function into `parseCallback`:

Old: 100ms
New: 110ms

**Comment by vlakoff** [comment]
May I suggest you this alternative to benchmark?

    public static function parseCallback($callback, $default)
    {
        $parts = explode('@', $callback, 2);

        return isset($parts[1]) ? $parts : [$callback, $default];
    }

**Comment by vlakoff** [comment]
Have just done some quick and dirty benches.
PHP 5.5, local machine, 500,000 iterations, results in seconds.

* method 1: current method
* method 2: PR using regex
* method 3: replacing `static::contains` with `strpos() !== false`
* method 4: my above method, directly exploding

with an `@` in haystack:

    1.4969999790192
    2.2220001220703
    0.92899990081787
    0.75600004196167

without `@` in haystack:

    1.1300001144409
    1.5139999389648
    0.56799983978271
    0.78900003433228

So:
* (3) and (4) are ~1.5-2x faster than current (1).
* When there is an `@` found (4) is best, otherwise (3) performs better. I suppose there should be an `@` in most of the use cases.
* (2) is always the slowest, slower than current code.

Additional note: `parseCallback()` has been added in https://github.com/laravel/framework/commit/27102ee1741d0b5efff6ae4173e7da43873f94ae, and never used in Laravel itself.

**Comment by vlakoff** [comment]
TL;DR:
* PR worsens performances.
* Current code is readable, quite fast already and probably not a bottleneck.
* Don't touch.
